services/blockchain-service/blockchain-service-base.js

```diff
@@ -488,6 +488,24 @@ class BlockchainServiceBase {
         return gasPrice;
     }
 
+    async getWalletBalances(blockchain) {
+        const web3Instance = await this.getWeb3Instance(blockchain);
+        const publicKey = await this.getPublicKey(blockchain);
+
+        const blockchainTokenBalance = await web3Instance.eth.getBalance(publicKey);
+        const tracBalance = await this.callContractFunction(
+            'Token',
+            'balanceOf',
+            [await this.getPublicKey(blockchain)],
+            blockchain,
+        );
+
+        return {
+            blockchainToken: blockchainTokenBalance,
```

Comment:
Let me know if you have suggestion for better naming or to separate this into two functions